jfix-zookeeper/src/main/java/ru/fix/zookeeper/lock/PersistentExpiringLockManager.java

```diff
@@ -148,36 +138,155 @@ public void release(LockIdentity lockId) {
         }
     }
 
+    @Override
+    public void close() {
+        locksContainer.close();
+        lockProlongationScheduler.close();
+    }
+
     /**
-     * @param lockId
-     * @param lock
-     * @return false in case of Exception of lock expiration or losing lock ownership
+     * Contains the locks and provide synced access to remove, prolong and getState operations under locks
      */
-    private boolean checkAndProlongLockIfRequired(LockIdentity lockId, PersistentExpiringDistributedLock lock) {
-        try {
-            logger.debug("Check and prolong lockId={}", lockId);
-            return lock.checkAndProlongIfExpiresIn(
-                    config.get().getLockAcquirePeriod(),
-                    config.get().getExpirationPeriod());
-        } catch (Exception e) {
-            logger.error("Failed to checkAndProlongIfExpiresIn persistent locks with lockId {}", lockId, e);
-            return false;
+    static class ActiveLocksContainer implements AutoCloseable {
+        /**
+         * remove, prolong and getState operations MUST BE synced under {@link #globalLock}
+         */
+        private final ConcurrentMap<LockIdentity, LockContainer> locks = new ConcurrentHashMap<>();
+        private final Object globalLock = new Object();
```

Comment:
Ивините, что влезаю, но может быть нам вместо globalLock и synchronized использовать ReentrantLock?
Вроде по производительности лучше, но это еще доказать нужно. 
Точно, что с помощью ReentrantLock мы можем использовать fair ordering policy, чтобы не получилось, что пока работает schedule все ждут.

Reply:
> Вроде по производительности лучше, но это еще доказать нужно

Sounds like you are giving a suggestion, but burden of research for synchonized vs ReentrantLock performance question you laying down on the developer shoulders. Providing link to proper research would be nice. 
As to the question: any performance gain would be insignificant for out scheduler that operates about tens of seconds with relatively small amount of locks.

> мы можем использовать fair ordering policy

Fair ordering sounds good, especially since distance between releasing and acquiring lock inside loop is short.

Reply:
Fixed to ReentrantLock